### ISSUE TITLE
PHPUnit genrates invalid mock code when false is in a union e.g. false|int

### ISSUE THREAD

**Issue by olvlvl** [sebastianbergmann/phpunit#4735]
| Q                   | A
| --------------------| ---------------
| PHPUnit version     | 9.5.6
| PHP version         | 8.0.8
| Installation Method | Composer

#### Summary

PHPUnit generates invalid mock code when the type `false` is used in a union e.g. `false|int`.

#### Current behavior

The following code is generated:

```php
declare(strict_types=1);

class false|int
{
}

class Mock_false|int_189b4275 extends false|int implements PHPUnit\Framework\MockObject\MockObject
{
    use \PHPUnit\Framework\MockObject\Api;
    use \PHPUnit\Framework\MockObject\Method;
    use \PHPUnit\Framework\MockObject\MockedCloneMethod;
}
```

#### How to reproduce

Here's an example class:

```php
<?php

namespace Dummy;

class Sample
{
    public function doSomething(): false|int
    {
        return false;
    }
}
```

Here's an example test:

```php
<?php

namespace Tests\Dummy;

use Dummy\Sample;
use PHPUnit\Framework\TestCase;

class SampleTest extends TestCase
{
    public function testExec()
    {
        $mock = $this->createMock(Sample::class);

        $mock->doSomething("hello");
    }
}
```

#### Expected behavior

The code for the mock is generated correctly.

#### Note

Using `bool|int` fixes the issue, but it changes the meaning of the function, because it will never return `true`. Besides, `false|int` is a valid unit type.

**Comment by sebastianbergmann** [comment]
Thank you for reporting this. This only happens when you create a test double and then do not configure a return value for a method that returns `*|false`.

**Comment by olvlvl** [comment]
The following code works indeed:

```php
class SampleTest extends TestCase
{
    public function testExec()
    {
        $mock = $this->createMock(Sample::class);
        $mock->method('doSomething')->willReturn(false, 123);

        $this->assertFalse($mock->doSomething("hello"));
        $this->assertEquals(123, $mock->doSomething("hello"));
    }
}
```

But how's one supposed to know? Could PHPunit display an error message instead of generating invalid code?

**Comment by sebastianbergmann** [comment]
I do not understand what you are asking. The issue you reported has been fixed in fe06cbef9eac32b0ed9a612444e5622e90072033.

**Comment by olvlvl** [comment]
Sorry, I missed this bit:

<img width="478" alt="image" src="https://user-images.githubusercontent.com/155998/126137083-242358e6-8327-4dae-96ab-0ed55b802daa.png">
